### PR TITLE
Chose service accounts dialog

### PR DESCRIPTION
**Pull request by inlined**
Part 1 of 4 for secrets:set

### CODE REVIEW

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:32]: Let's include a line comment with an example of a raw backend.name value

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:36]: If we're sorting, consider adding a second sort column for name (will help with users with multi-regional backends)

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:88]: Do we want to tell the user the name of the backend?

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:99]: typo in comparison

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:102]: Alternatively, you can do:

new Set(metadata.map(val => JSON.stringify(val.accounts)).size === 1

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:101]: firstBackendServiceAccounts?

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [inline src/apphosting/secrets/dialogs.ts:108]: Is the intention of this method to be shared between `set` and `grantAccess`? If so, this messaging feels overly verbose for the `grantAccess` usecase

**Review by tonyjhuang — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some comments

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:10]: Done

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:16]: Done

Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:16]: That function is not forwards compatible with explicit SAs and I'm not sure how to use the map of SAs when there will soon only be one.

My intention was to:
1. Finish the set command reasonably independently
2. Migrate all the existing source to appropraite locations
3. Reconcile the various commands

I still need to check with Abhi why the viewer permission is needed for Build. That may change things in the future.

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:32]: Done

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:36]: That's what the next condition does (`return left.id.localeCompare(right.id)`)

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:88]: If they only have one backend that seems gratuitous, no?

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:99]: Done

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:108]: I'm not sure I agree, but this also sounds like a discussion we can have once we're ready to reevaluate `secrets:grantaccess` and whether it should support the dialog.

File: src/gcp/iam.ts
Comment (inlined) [reply src/gcp/iam.ts:8]: Prevoiusly gcb had the helpers for the default compute service agent as well. Historically we've thrown common IAM concerns (where possible) in the IAM folder. We could create a `src/gcp/computeengine` file that literally only has one const if you desire.

If we move into gcb and gce files, I would suggest that we shorten the functions to `getDefaultServiceAgent`

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts:21]: Hmmm... Given the fact that the build account needs an extra role I see what this was doing. I'll reconcile. This will be annoying and I need to go rehearse for I/O, so I'd like to get sign-off on what I have with a pinky promise to reconcile/fix this when I continue with the change tonight.

Comment (tonyjhuang) [reply src/apphosting/secrets/dialogs.ts:21]: sg, lets ship this for now and we can revisit later, esp much of this will change with the pending service account API updates

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [reply src/apphosting/secrets/dialogs.ts:134]: sg. Personally, I would assume users would understand this point when they call `secrets:grantAccess`, but happy to re-eval later

File: src/gcp/iam.ts
Comment (tonyjhuang) [reply src/gcp/iam.ts]: Separating out by service files feels more extensible to me but I'll leave it up to you

File: src/apphosting/secrets/dialogs.ts
Comment (tonyjhuang) [reply src/apphosting/secrets/dialogs.ts:112]: We also reconfirm the backend in `backends:delete` if a backend id is passed in but the `location` is omitted. I'll leave it up to you here

**Review by tonyjhuang — APPROVED** [review]
Left a few followup comments but overall lgtm thanks for the change!

File: src/apphosting/secrets/dialogs.ts
Comment (inlined) [reply src/apphosting/secrets/dialogs.ts]: Oh, forgot to mention here that `every` short circuits but `map` does not. It's a trivial optimization, but seems correct to use in this case.

**Pull request merged**
